### ISSUE TITLE
Same event being displayed twice on dashboard after clicking "show more events"

### ISSUE THREAD

**Issue by jesseallhands** [Couchers-org/couchers#2978]
The following event shows up twice on the dashboard. It shows up on the dashboard once initially but when you click "show more events" it shows the event twice. I confirmed it is indeed the same event (not duplicate events):

https://couchers.org/event/1391/rock-im-park-von-fra

![image](https://user-images.githubusercontent.com/79384268/170982826-6b150f70-bc96-4b47-9b73-58793fce242b.png)

**Comment by guillermodlpa** [comment]
I've investigated this, and replicated in the live site. tl;dr: the frontend is correct. The problem appears to be on the pagination mechanism used by the backend, and this situation can happen depending on the end time of events.

---

On the frontend side, things work correctly:

1. The first 2 events are fetched. The response includes a `nextPageToken`
2. The next page request is made using the `nextPageToken`, and also includes the page size in the request payload.

In the live site, right now, the 2nd page however includes again one event from the 1st page, why?

<img width="941" alt="Screen Shot 2022-05-31 at 11 51 23" src="https://user-images.githubusercontent.com/1557348/171153734-a816d10b-a7d0-4533-9a6c-ad5f33855320.png">

---

Looking in the backend code, I see that the `nextPageToken` is the end time of the 1st event of the following page. And then, when fetching the next page, that token is used to fetch events that end after that time.

So with these events:

1. Event A (83rd Weekly Online Event!, ends on May 31 at 3pm)
2. Event B (Rock im Park von FRA, **ends on Jun 5 at 1pm**)
3. Event C (Cya later picnic Kaunas, **ends on Jun 4 at 1pm**)

The 1st fetch would return in its `nextPageToken` the end time of event C. **The 2nd fetch would return events ending at the same time or later than Event C**. The reason why Event B is returned again is because Event B ends later than Event C.

When looking at how the event's end time is used for queries, it makes sense:

<img width="1019" alt="Screen Shot 2022-05-31 at 12 14 54" src="https://user-images.githubusercontent.com/1557348/171151009-78cba87f-bf64-4d61-bda5-4bfbe19bf3a3.png">

Results are sorted by event start, so that's why the first page includes Event A and B.

--

**Recommended backend fix:** update the endpoint to use start time for filtering, not end time.

Potentially though, we should use another cursor different than time, because if there were more events starting at the same time than the page size (2), this mechanism won't work and the same events would be returned all the time.

Since this appears to be a backend fix, and I don't have the backend env set up locally nor experience with Python, I'll unassign this from myself. I'll assign it to myself again in the future if I have time to set it up and learn new things while fixing this! 👍

**Comment by Vinalti** [comment]
You are right I think, alone, the time doesn't seem to be a good cursor to iterate over events, as you said, we could have many events starting/ending at the same time, nevertheless we could use the ID in order to filter out the last returned event, no ? 

Here is the solution I would propose :
- We add the `event_id` (on top of the start-time or end-time) of the last returned event in the cursor (or next to it). Then we can fetch the events depending on start or end time and discard any event until the ID of the last event. 
 
I am honnestly not very familiar with the code so maybe my solution doesn't work as is, but conceptually, I would propose to implement it kinda like that: 

```py
last_id = request.last_event_id  # We get the ID of the last retrieved event in the request
for i, event in occurences:
    if event.event_id == last_event_id:
        occurences = occurences[i+1:]  # We iterate over the occurences, and if we find the id we slice occurence from it.
        break
```
And we can retrieve it like this : 
```py
page_token, last_id = (None, None)
if (request.page_token):
    page_token, last_id = request.page_token.split('+')
    page_token = dt_from_millis(int(page_token))
    last_id = int(last_id)
else:
    page_token = now()
```
And return the next cursor
```py
last_event = occurrences[-1]
next_page_token = str(millis_from_dt(last_event.end_time))+"+"+last_event.event_id if len(occurrences) > page_size else None
```

Current code :
https://github.com/Couchers-org/couchers/blob/d04e9dbe68d0bebb3a6086fdc169988186571acc/app/backend/src/couchers/servicers/events.py#L757-L772

**Comment by aapeliv** [comment]
Fixed in https://github.com/Couchers-org/couchers/pull/9329
